test/config/database.yml

```diff
@@ -0,0 +1,9 @@
+default: &default
+  adapter: sqlite3
+  pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
+  timeout: 5000
+
+test:
+  <<: *default
+  database: db/test.sqlite3
+
```

Comment:
Do we have to add a db to make the test cases work? We could have a model without it being backed by a db

Reply:
The reason for the db was to get the tests to pass. ActionText requires a db.  I agree that we could just provide instructions for each user to add this functionality.  Since Rails 6.1 is supposed to be released incorporating ViewComponents would there be a way to add an installer that would add the required code to the ApplicationComponent should they require ActionText in ViewComponents?

Reply:
As per your suggestion I tried out creating an ApplicationComponent as follows:

`require "action_text/engine"
class ApplicationComponent < ViewComponent::Base
  include ActionText::Engine.helpers

  def main_app
     Rails.application.class.routes.url_helpers
  end
end`

Then my components inherit from this class.  This works.  However, there is still an issue with getting ViewComponents  to work with ActionText outside of Rails. ActionText has a file called tag_helper.rb which references rails:

```
    def rich_text_area_tag(name, value = nil, options = {})
      options = options.symbolize_keys

      options[:input] ||= "trix_input_#{ActionText::TagHelper.id += 1}"
      options[:class] ||= "trix-content"

      options[:data] ||= {}
      options[:data][:direct_upload_url] = main_app.rails_direct_uploads_url
      options[:data][:blob_url_template] = main_app.rails_service_blob_url(":signed_id", ":filename")

      editor_tag = content_tag("trix-editor", "", options)
      input_tag = hidden_field_tag(name, value, id: options[:input])

      input_tag + editor_tag
    end
  end
```

So I don't see how to get ActionText working with ViewComponent without the Rails dependency.

Reply:
I'm not disputing that you need Rails to use ViewComponent with or without ActionText. My question is what's the minimum we need with this change to support ActionText and to test it effectively. Maybe it will become more clear once you have fleshed out the test cases.

Reply:
@jonspalmer I just added conditionals to check if ActionText is loaded and if it is loaded then load the necessary libraries to support it.

 I don't know if they way I have done this is best practices, I'm just trying to find a solution and will gladly follow any advice.  I understand that the minimal changes to ViewComponent required is the goal.

Reply:
I moved the support for action text to a module that checks if action text has been loaded.